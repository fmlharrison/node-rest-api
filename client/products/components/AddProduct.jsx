import React, { Component } from "react";

class AddProduct extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: "",
      price: ""
    };
  }

  updateName = (e) => {
    const name = e.target.value;
    this.setState({ name })
  }

  updatePrice = (e) => {
    const price = e.target.value;
    this.setState({ price })
  }

  render() {
    return (
      <div>
        <form>
          <label for="product-name">Product Name</label>
          <br />
          <input type="text" id="product-name" name="name" value={this.state.name} onChange={this.updateName}/>
          <label for="product-price">Price</label>
          <br />
          <input type="text" id="product-price" name="price" value={this.state.price} onChange={this.updatePrice}/>
          <br />
          <input type="submit" value="Add Product"/>
        </form>
      </div>
    );
  }
}

export default AddProduct;
