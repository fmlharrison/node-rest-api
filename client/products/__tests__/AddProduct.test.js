import React from "react";

import AddProduct from "../components/AddProduct";

test("renders a form to add a new product", () => {
  const component = shallow(<AddProduct />);
  expect(component.state().name).toBe("");
  expect(component.state().price).toBe("");
  expect(component).toMatchSnapshot();
});

test("Changes the state of the component when user inputs text in form", () => {
  const component = shallow(<AddProduct />);

  component
    .find("#product-name")
    .simulate("change", { target: { value: "Playstation" } });

  component
    .find("#product-price")
    .simulate("change", { target: { value: "550" } });

  expect(component.state().name).toBe("Playstation");
  expect(component.state().price).toBe("550");
});
