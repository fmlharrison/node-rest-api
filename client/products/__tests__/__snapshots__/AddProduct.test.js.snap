// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a form to add a new product 1`] = `
<div>
  <form>
    <label
      for="product-name"
    >
      Product Name
    </label>
    <br />
    <input
      id="product-name"
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      for="product-price"
    >
      Price
    </label>
    <br />
    <input
      id="product-price"
      name="price"
      onChange={[Function]}
      type="text"
      value=""
    />
    <br />
    <input
      type="submit"
      value="Add Product"
    />
  </form>
</div>
`;
